source/simulator/simulator_signals.cc

```diff
@@ -28,7 +28,8 @@ namespace aspect
 {
   // provide locations for the static variables of this class
   template <int dim>
-  boost::signals2::signal<void (ParameterHandler &)>  SimulatorSignals<dim>::declare_additional_parameters;
+  boost::signals2::signal<void (const unsigned int,
```

Comment:
It's a member function in a structure `SimulatorSignals<dim>` that depends on the dimension. The member function stores a pointer to the function. If you call the function pointed to, it doesn't know where you called it from, so the dimension is lost unless you specifically pass it as an argument.